### ISSUE TITLE
Helm chart required

### ISSUE THREAD

**Issue by jpds** [authelia/authelia#274]
The documentation at https://github.com/clems4ever/authelia/blob/master/example/kube/README.md should reference a Helm chart which is used to deploy Authelia.

The deployment shouldn't also statically hard-code other components (such as nginx-ingress controllers) as these will just grow stale with time (`0.13.0` is used in the manifests when `0.19.0` is out).

**Comment by clems4ever** [comment]
Hello @jpds , I agree that I should use a Helm chart. I just wanted to have a working version but it can definitely be improved. I also agree on the hard-coded tag but there is a problem though. I have no integration tests against a Kube cluster so far so bumping to a newer version might break the tuto... Templating the tuto is a bit risky right for now.

I'll try to address those issues though. Thank you for reporting.

**Comment by clems4ever** [comment]
@jpds , Helm chart is under construction as you can read in #354 .

**Comment by clems4ever** [comment]
@darkquasar , do you have a chart template you can publish somewhere so that I can help you out?

**Comment by darkquasar** [comment]
Hi mate, yes, I will share what I've done so far next week, I use Authelia as a subchart.

**Comment by tarioch** [comment]
If it's any help, I made a chart for the "development mode" (without ldap and redis) https://github.com/tarioch/helm/tree/master/charts/authelia

**Comment by clems4ever** [comment]
No pb @darkquasar.
Thanks @tarioch, I will have a look at it, it's a very good start for me to ramp up on Helm.

**Comment by darkquasar** [comment]
Hi Clement sorry for the late reply, please check here: https://github.com/darkquasar/socops/tree/dev/helm/socops/charts/authelia 

## Authelia Chart

This is a working example of the chart. Steps to reproduce:

1. Create a fresh Ubuntu 16 or 18 VM
2. Install microk8s (or minikube or whatever you use to test k8s, I will assume you use microk8s for this explanation). `snap install microk8s --classic`
2. Enable dashboard, dns, and ingress for the microk8s instance: `microk8s.enable storage ingress dns dashboard`. As you will see, the chart currently doesn't deploy an *ingress controller* because you can deploy it natively in microk8s, however, the choice to install one should be added to the chart. 
3. Download Helm binary and initialize Tiller.
4. The Authelia chart uses other sub-charts as dependencies (MongoDB, Redis, OpenLDAP), so we need to add this to the local Helm cache before we can install Authelia: `helm dependency update authelia` (or `socops/charts/authelia`, wherever the Authelia chart lives, point to that :) )
5. **IMPORTANT**: adjust the Authelia config in `./files/authelia-config.yml` to match your test ground, then also adjust the chart's `values.yaml` if required to match authelia-config.yml.
6. Finally just deploy the chart: `helm install -n authelia authelia` (**Note:** the first *authelia* is the name of the release, the 2nd is the name of the folder where we keep our raw chart files)

My chart includes **phpldapadmin** by default so that editing the contents of the openldap is a breeze. The **openldap** chart also comes pre-populated with values, check the main `values.yaml` which overrides the defaults of the openldap chart. To test the login use the following credentials: `jaguar01:password`. 

All the services are reachable via their service cluster IPs.


## Todo

The current chart needs some improvements:
* Add the option to install an ingress controller using a switch in values.yaml (something like *install_ingress: true/false*)
* Craft the ingress.yaml file so that instead of having two hard-coded ingresses, they can be dynamically generated by passing the config to values.yaml (it will require some "range" iteration)
* Craft the secrets.yaml file so that instead of having two hard-coded TLS secrets for the two hard-coded ingresses, they can dynamically generate secrets based on the amount of sites (ingressess) we want to protect

**Comment by clems4ever** [comment]
Awesome! Just an additional info, in the integration tests I also need to deploy a test cluster and it's done using https://github.com/kubernetes-sigs/kind. The binary is automatically downloaded when one runs `source bootstrap.sh`.

@darkquasar , you can do a PR with what you already have if you wish. You can take this opportunity to include your recipe in the doc somewhere. @darkquasar , @tarioch , I think we can start from there and improve it. I can help building a suite to ensure non-regression on that part if needed.

What do you guys think about that?

**Comment by darkquasar** [comment]
@clems4ever yeap, I will submit a PR over the next few days and we will take it from there.

**Comment by clems4ever** [comment]
A PR is open for v3. This will be the base for v4 when merged.
https://github.com/clems4ever/authelia/pull/458

**Comment by nightah** [comment]
Closing in favour of #575 given that v3 is obsolete.
